elements/lisk-blocks/src/state_store/account_store.ts

```diff
@@ -156,10 +136,17 @@ export class AccountStore {
 	public find(
 		fn: (value: Account, index: number, obj: Account[]) => unknown,
 	): Account | undefined {
-		return this._data.find(fn);
+		const foundAccount = this._data.find(fn);
+		if (!foundAccount) {
+			return undefined;
+		}
+
+		return foundAccount;
 	}
 
 	public set(primaryValue: string, updatedElement: Account): void {
+		const updatedElementObj = new Account(updatedElement);
```

Comment:
if it's setting the account, there is no need to create new instance

Reply:
Sometimes the object coming from the framework is not an Account object when we are running functional tests so I will put a condition to cover that